src/containers/Surah/index.js

```diff
@@ -340,18 +340,20 @@ export default class Surah extends Component {
   }
 
   renderLines() {
-    const { lines, options } = this.props;
+    const { lines, options, ayahs } = this.props;
     const keys = Object.keys(lines);
+    const ayahsKeys = Object.keys(ayahs);
 
     return keys.map((lineNum, index) => {
       const nextNum = keys[index + 1];
       const pageNum = lineNum.split('-')[0];
       const line = lines[lineNum];
+      const lastAyahInfo = ayahs[ayahsKeys[ayahsKeys.length - 1]]; // last ayah
```

Comment:
I found a bug here... since you are using the last ayah in the ayahs, when lazy loading, this will take the last ayah and apply it to all `<PageBreak />` components... so they will all render the last ayah's hizb, juz and page number, although they are not. See:

![image](https://cloud.githubusercontent.com/assets/3749095/16572533/54ae5dde-421e-11e6-8011-8ec3488aa4e6.png)